src/Analyzers/CSharp/CodeFixes/UseThrowExpression/UseThrowExpressionCodeFixProvider.cs

```diff
@@ -52,14 +55,37 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 var ifStatement = root.FindNode(diagnostic.AdditionalLocations[0].SourceSpan);
                 var throwStatementExpression = root.FindNode(diagnostic.AdditionalLocations[1].SourceSpan);
                 var assignmentValue = root.FindNode(diagnostic.AdditionalLocations[2].SourceSpan);
+                var expressionStatement = assignmentValue.GetAncestor<ExpressionStatementSyntax>();
+                var identifierName = expressionStatement.DescendantNodes().ElementAt(1);
```

Comment:
I think it should be possible to find the indentifier directly, but we may not need this bit entirely (see other comments)